bitmapcontainer.go

```diff
@@ -314,12 +316,14 @@ func (bc *bitmapContainer) iaddReturnMinimized(i uint16) container {
 
 func (bc *bitmapContainer) iadd(i uint16) bool {
 	x := int(i)
-	previous := bc.bitmap[x/64]
-	mask := uint64(1) << (uint(x) % 64)
-	newb := previous | mask
-	bc.bitmap[x/64] = newb
-	bc.cardinality += int((previous ^ newb) >> (uint(x) % 64))
-	return newb != previous
+	previous := bc.bitmap[x>>log2WordSize]
+	mask := uint64(1) << (uint(x) & 63)
+	if newb := previous | mask; newb != previous {
```

Comment:
Here you are turning a potentially branchless routine into a branchy routine. This can have far reaching consequences which are difficult to assess using a simple benchmark.